bandit/bandit.go

```diff
@@ -163,7 +163,7 @@ func (o *BanditDialer) DialContext(ctx context.Context, network, addr string) (n
 
 	countryCode, country, city := dialer.Location()
 	previousStats := o.statsTracker.Latest()
-	if previousStats.CountryCode != "" && previousStats.CountryCode != countryCode {
+	if previousStats.CountryCode == "" || previousStats.CountryCode != countryCode {
```

Comment:
@reflog 

So the problem here is we were checking  `previousStats.CountryCode != ""` but using old code previousStats.CountryCode it always remains null because we were setting o.statsTracker.SetActiveProxyLocation. only when `previousStats.CountryCode != ""`  and without setting `o.statsTracker.SetActiveProxyLocation` previous state is always going to be null

Reply:
So new code checks if `previousStats.CountryCode == ""` is null then set location so we on next call we will get the previous state not null and it will check for `previousStats.CountryCode != countryCode``